### DISCUSSION TITLE
setrc usage

### DISCUSSION THREAD

**Discussion by chros098** [The-OpenROAD-Project/OpenROAD-flow-scripts#2429] [Q&A] [answered]
I would like to ask how to use setRC. At which step is it used? And what is the basis for judging the units of resistance and capacitance in the setRC file? Is it based on the units in the first lib file read? How are the numbers in setrc obtained? According to the relevant instructions, they are obtained from the lef file. However, the values ​​in the setrc files of several pdk files provided on openroad do not correspond to the lef files.

**Comment by maliberty** [comment]
There are two commands in a typical setRC, set_layer_rc and set_wire_rc.  The former defines the estimated res/cap per unit length and is used anytime we need the parasitics for a net except for final signoff where we do RCX.  The later is used when we don't yet have layer assignment and is used to set an assumed typical layer (eg Steiner trees pre-global routing).

Often the DRM will give values that cvan be used for set_layer_rc.  There is a more involved process where you can run a bunch of designs and statistically extract average values per layer.

**Comment by chros098** [comment]
Do you mean that the value in setRc is the average value after running several times, not directly obtained from lef?

**Reply by maliberty** [reply]
If the value is LEF is acceptable then you don't need to specify a set_wire_rc.  Often it is some junky placeholder value.

**Comment by chros098** [comment]
What exactly does DRM mean?

**Reply by maliberty** [reply]
DRM = Design Rule Manual

**Comment by chros098** [comment]
I understand. But I didn't find the relevant value in DRM. Isn't this value consistent with the value in lef?

**Reply by maliberty** [reply]
It depends on who produces the PDK- it may or may not be.  I can't speak to your DRM but it is common.

**Comment by chros098** [comment]
Thank you for your answer. I am using openroad and want to add a new process. I need to write a new setrc file, but I don't know how to write the specific values. If the DRM file does not have it, what other methods and software can I use?

**Reply by maliberty** [reply]
If the LEF has a value you can try it and see if the results seem ok.  If not you can setup RCX (which you'll need to do anyways) and then extract some designs to get typical values.  

@precisionmoon do you have a write up on the set_wire_rc methodology you used recently?

**Reply by precisionmoon** [reply, marked as answer]
I'm attaching a prelim methodology doc.   Your comments are welcome.
[Correlating RC values for PDKs.pdf](https://github.com/user-attachments/files/17289399/Correlating.RC.values.for.PDKs.pdf)

**Reply by chros098** [reply]
thank， I'm trying

**Comment by chros098** [comment]
rcx has been set, what values ​​should be referenced in the rcx file?
